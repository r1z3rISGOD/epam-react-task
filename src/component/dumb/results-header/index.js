import {ResultsHeader} from "./results-header.js"
export {ResultsHeader}
