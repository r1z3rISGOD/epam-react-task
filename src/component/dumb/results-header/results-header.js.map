{"version":3,"file":"results-header.js","sourceRoot":"","sources":["results-header.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,iCAA8B;AAC9B,6DAA0D;AAC1D,kDAA+C;AAC/C,gDAA6C;AAC7C,gDAA6C;AAC7C,sDAAmD;AAY5C,IAAM,aAAa,GAAkC,UAAC,EAQ5D;QAPC,YAAY,kBAAA,EACZ,UAAU,gBAAA,EACV,aAAa,mBAAA,EACb,gBAAgB,sBAAA,EAChB,UAAU,gBAAA,EACV,cAAc,oBAAA,EACd,UAAU,gBAAA;IAEV,QAAQ,UAAU,EAAE;QAClB,KAAK,CAAC;YACJ,OAAO,CACH,6BAAK,SAAS,EAAC,gBAAgB;gBAC3B,oBAAC,0BAAW;oBACR,oBAAC,8BAAa,IAAC,EAAE,EAAE,IAAI;wBACnB,oBAAC,4BAAY,IACT,YAAY,EAAE,YAAY,GAAG,CACrB;oBAChB,oBAAC,0BAAW,IACR,UAAU,EAAE,UAAU,EACtB,aAAa,EAAE,aAAa,EAC5B,cAAc,EAAE,cAAc,EAC9B,UAAU,EAAE,UAAU,GACxB,CACQ,CACZ,CACT,CAAA;QACH,KAAK,CAAC;YACJ,OAAO,CACH,6BAAK,SAAS,EAAC,gBAAgB;gBAC3B,oBAAC,0BAAW;oBACR,oBAAC,gCAAc,IAAC,gBAAgB,EAAE,gBAAgB,GAAG,CAC3C,CACZ,CACT,CAAA;QACH;YACE,OAAO,CACH,6BAAK,SAAS,EAAC,gBAAgB,GAAE,CACpC,CAAA;KACJ;AACH,CAAC,CAAA;AAxCY,QAAA,aAAa,iBAwCzB"}