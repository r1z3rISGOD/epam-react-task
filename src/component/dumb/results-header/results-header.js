import React from 'react'
import { ErrorBoundary } from '../../smart/error-boundary'
import { ResultsCount } from '../results-count'
import { ResultsSort } from '../results-sort'
import { ResultsData } from '../results-data'
import { ResultsRelated } from '../results-related'
import './results-header.scss'

export const ResultsHeader = (props) => {
  const { resultsCount, isOpened, active, setActive, searchData, searchSetting, openedFilmsGenre } = props

  if (resultsCount > 0 && !isOpened) {
    return (
            <div className="results-header">
                <ResultsData>
                    <ErrorBoundary>
                        <ResultsCount
                            resultsCount={resultsCount}
                        />
                    </ErrorBoundary>
                    <ResultsSort
                        active={active}
                        setActive={setActive}
                        resultsCount={resultsCount}
                        searchData={searchData}
                        searchSetting={searchSetting}
                    />
                </ResultsData>
            </div>
    )
  } else if (isOpened) {
    return (
            <div className="results-header">
                <ResultsData>
                    <ResultsRelated openedFilmsGenre={openedFilmsGenre}/>
                </ResultsData>
            </div>
    )
  } else {
    return (
            <div className="results-header"/>
    )
  }
}
