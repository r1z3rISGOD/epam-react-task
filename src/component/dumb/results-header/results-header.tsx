import * as React from 'react'
import './results-header.scss'
import { ErrorBoundary } from '../../smart/error-boundary'
import { ResultsCount } from '../results-count'
import { ResultsSort } from '../results-sort'
import { ResultsData } from '../results-data'
import { ResultsRelated } from '../results-related'

interface ResultsHeaderProps {
    resultsCount: number
    searchData: string
    searchSetting: string
    openedFilmsGenre: string
    headerType: number
    sortingResults: any
    sortVision: number
}

export const ResultsHeader : React.FC<ResultsHeaderProps> = ({
  resultsCount,
  searchData,
  searchSetting,
  openedFilmsGenre,
  headerType,
  sortingResults,
  sortVision
}) => {
  switch (headerType) {
    case 1:
      return (
          <div className="results-header">
              <ResultsData>
                  <ErrorBoundary ok={true}>
                      <ResultsCount
                          resultsCount={resultsCount}/>
                  </ErrorBoundary>
                  <ResultsSort
                      searchData={searchData}
                      searchSetting={searchSetting}
                      sortingResults={sortingResults}
                      sortVision={sortVision}
                  />
              </ResultsData>
          </div>
      )
    case 2:
      return (
          <div className="results-header">
              <ResultsData>
                  <ResultsRelated openedFilmsGenre={openedFilmsGenre}/>
              </ResultsData>
          </div>
      )
    default:
      return (
          <div className="results-header"/>
      )
  }
}
