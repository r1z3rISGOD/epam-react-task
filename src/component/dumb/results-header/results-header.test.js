import { ResultsHeader } from './results-header.tsx'
import React from 'react'

function f () {
  return 1
}

function f1 () {
  return 1
}

describe('results header tests', () => {
  it('results header test 1', () => {
    const comp = shallow(<ResultsHeader headerType={1} openedFilmsGenre='drama' resultsCount={10} searchData='avenger' searchSetting='title' sortingResults={f} sortVision={f1}/>)
    expect(comp).toMatchSnapshot()
  })
  it('results header test 2', () => {
    const comp = shallow(<ResultsHeader headerType={2} openedFilmsGenre='drama' resultsCount={10} searchData='avenger' searchSetting='title' sortingResults={f} sortVision={f1}/>)
    expect(comp).toMatchSnapshot()
  })
  it('results header test 3', () => {
    const comp = shallow(<ResultsHeader headerType={0} openedFilmsGenre='drama' resultsCount={10} searchData='avenger' searchSetting='title' sortingResults={f} sortVision={f1}/>)
    expect(comp).toMatchSnapshot()
  })
})
