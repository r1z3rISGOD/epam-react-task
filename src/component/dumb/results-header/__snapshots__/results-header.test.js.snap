// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results header tests results header test 1 1`] = `
<div
  className="results-header"
>
  <ResultsData>
    <ErrorBoundary
      ok={true}
    >
      <ResultsCount
        resultsCount={10}
      />
    </ErrorBoundary>
    <ResultsSort
      searchData="avenger"
      searchSetting="title"
      sortVision={[Function]}
      sortingResults={[Function]}
    />
  </ResultsData>
</div>
`;

exports[`results header tests results header test 2 1`] = `
<div
  className="results-header"
>
  <ResultsData>
    <ResultsRelated
      openedFilmsGenre="drama"
    />
  </ResultsData>
</div>
`;

exports[`results header tests results header test 3 1`] = `
<div
  className="results-header"
/>
`;
